### PR TITLE
Added Java Implementation for Family Tree [833] (bronze)

### PR DESCRIPTION
**Pull request by ChuangSheep**
All test cases passed. 
<s>I'm gonna also write an implementatino for Sol1. </s>   **Done.**

### CODE REVIEW

**Review by brad182 — APPROVED** [review]

**Comment by ChuangSheep** [comment]
emmm, any extra changes wished?

**Review by dongliu0426 — APPROVED** [review]
would prefer kattio over BufferedReader/PrintWriter but ok

File: solutions/usaco-833.mdx [hunk 1]
```diff
@@ -103,6 +103,103 @@ else:
...
+			if (child.equals(pair[1]))
+				return pair[0];
+		}
+		return null;
+	}
+
+	// calculate the distance between the given node start and the given end
+	// if end is indeed an ancestor of start, otherwise, return -1
+	private static int getAncestorDistance(String end, String start, 
+	String[][] pairs) {
```
Comment (maggie-j-liu) [inline solutions/usaco-833.mdx:188]: formatting here looks kind of weird, not sure if this is better?
```
	private static int getAncestorDistance(
		String end,
		String start, 
		String[][] pairs
	) {
```

Comment (ChuangSheep) [reply solutions/usaco-833.mdx:188]: yo Im gonna change it.

**Review by maggie-j-liu — APPROVED** [review]

**Pull request merged**